### DISCUSSION TITLE
Prefix don't work in PlayCDN

### DISCUSSION THREAD

**Discussion by ribelo** [tailwindlabs/tailwindcss#9935] [Help] [answered]
**What version of Tailwind CSS are you using?**

CDN. Newest. Probably 3.2.4

**What build tool (or framework if it abstracts the build tool) are you using?**

Not applicable.

**What version of Node.js are you using?**

Not applicable.

**What browser are you using?**

Every.

**What operating system are you using?**

Btw, I use Arch.

**Reproduction URL**

Not applicable.

**Describe your issue**

As in the title. In playCDN the prefix setting does not work.
![2022-11-25-101544_grim](https://user-images.githubusercontent.com/1815898/203944445-d1bf1b4e-2cf7-4381-911c-5400f8a752aa.png)
![2022-11-25-101527_grim](https://user-images.githubusercontent.com/1815898/203944451-4d6255dc-4fb0-46c2-ae51-1451b1a29615.png)
![2022-11-25-101458_grim](https://user-images.githubusercontent.com/1815898/203944455-5174bcd5-2579-4fa1-be68-322a5f9d2b68.png)

Post Scriptum.

Yes, I know, the rest of the settings are not needed in playCDN, but we only want to use playCDN in developer mode due to the problem of using tailwindcss/postcss in large projects written in clojurescript. Topic for another thread.

**Comment by adamwathan** [comment, marked as answer]
Can't reproduce, seems to work here:

https://jsfiddle.net/oj9scwvq/3/

Converting this to a discussion since it works as expected in my testing and no reproduction of the issue was provided so there's no action for us to take here, but if you can provide a reproduction happy to look into it.

**Reply by ribelo** [reply]
The problem was obviously on my side. Sorry for bothering you and thanks for the instant reply. :)

**Comment by Scion-Kin** [comment]
Having the same issue right now.
Is tailwind config no longer supported via cdn?

**Reply by rozsazoltan** [reply]
From v4 onward, it is no longer enabled by default, and there is no backwards compatibility solution in the CDN. The Play CDN is not meant for development! Do not use it for production. It should only be used for quick prototypes or reproductions.

* https://github.com/tailwindlabs/tailwindcss/discussions/9935#discussioncomment-15230833

Use one of the NPM packages (CLI, Vite, or PostCSS plugins) or the Standalone version (which doesn't require Node.js or NPM). TailwindCSS should be used strictly during development - it is a developer tool. For production, only the compiled CSS should be shipped.

**Comment by rozsazoltan** [comment]
> TailwindCSS is a development tool. The Play CDN exists solely for creating quick reproductions, but it's absolutely the wrong approach to use it in a real production application:
> * If the CDN becomes unavailable (for example, Cloudflare recently had an incident), your app will go down until it comes back (potentially for several hours).
> * On every load, the Tailwind engine unnecessarily rebuilds the exact same CSS that it already built the previous 1000, 2000, 3000 times - completely wasted work.

References:
* [Stop using Play CDN for production](https://stackoverflow.com/a/79832376/15167500)
* https://stackoverflow.com/a/79772442/15167500
* https://stackoverflow.com/a/79805241/15167500

**Comment by rozsazoltan** [comment]
Do not use the Play CDN in active products. Only for minimal reproductions: https://github.com/tailwindlabs/tailwindcss/discussions/9935#discussioncomment-15230833

If you need a prefix in such a reproduction, you can achieve it like this from v4: (**do not use for production**)
```html
<script src="https://cdn.jsdelivr.net/npm/@tailwindcss/browser@4"></script>
<style type="text/tailwindcss">
@import "tailwindcss" prefix(tw);

@theme {
  --color-clifford: #da373d;
}
</style>

<h1 class="tw:text-3xl tw:font-bold tw:underline tw:text-clifford">
  Hello world!
</h1>
```

* https://tailwindcss.com/docs/upgrade-guide#using-a-prefix
* https://tailwindcss.com/docs/styling-with-utility-classes#using-the-prefix-option
* [How to add tw- prefix to Tailwind CSS v4?](https://stackoverflow.com/a/79411170/15167500)

Other:
* [What's breaking changes from v4?](https://stackoverflow.com/a/79383884/15167500)
* [How to use custom color themes in TailwindCSS v4](https://stackoverflow.com/a/79499827/15167500)
